docs/topics/compatibility-guide-18.md

```diff
@@ -0,0 +1,638 @@
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: report a warning (or an error in the progressive mode) on such companion member access
+> - 1.9.0: raise the warning to an error,
+>   `-XXLanguage:-ProhibitAccessToEnumCompanionMembersInEnumConstructorCall` can be used to temporarily revert to the pre-1.8 behavior
+
+### Deprecate and remove `Enum.declaringClass` synthetic property
+
+> **Issue**: [KT-49653](https://youtrack.jetbrains.com/issue/KT-49653)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin allowed using the synthetic property `declaringClass` on `Enum` values produced from
+> the method `getDeclaringClass()` of the underlying Java class `java.lang.Enum` even though this method is not available
+> for Kotlin `Enum` type. Kotlin 1.9 will prohibit using this property, proposing to migrate to the extension property
+> `declaringJavaClass` instead
+>
+> **Deprecation cycle**:
+>
+> - 1.7.0: report a warning (or an error in the progressive mode) on `declaringClass` property usages,
+>   propose the migration to `declaringJavaClass` extension
+> - 1.9.0: raise the warning to an error,
+>   `-XXLanguage:-ProhibitEnumDeclaringClass` can be used to temporarily revert to the pre-1.9 behavior
+> - \>=1.10: remove `declaringClass` synthetic property
+
+### Deprecate the `enable` and the `compatibility` modes of the compiler option -Xjvm-default
+
+> **Issue**: [KT-46329](https://youtrack.jetbrains.com/issue/KT-46329)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6.20 warns about the usage of the `enable` and `compatibility` modes of the `-Xjvm-default` compiler option
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: introduce a warning on the `enable` and `compatibility` modes of the `-Xjvm-default` compiler option
+> - \>= 1.9: raise this warning to an error
+
+## Standard library
+
+### Warn about potential overload resolution change when `Range`/`Progression` starts implementing `Collection`
+
+> **Issue**: [KT-49276](https://youtrack.jetbrains.com/issue/KT-49276)
+>
+> **Component**: Core language / kotlin-stdlib
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: it is planned to implement the `Collection` interface in the standard progressions and concrete ranges
+> inherited from them in Kotlin 1.9. This could make a different overload selected in the overload resolution if there
+> are two overloads of some method, one accepting an element and another accepting a collection.
+> Kotlin will make this situation visible by reporting a warning or an error when such overloaded method is called
+> with a range or progression argument
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: report a warning when an overloaded method is called with the standard progression or its range inheritor argument
```

Comment:
```suggestion
> - 1.6.20: report a warning when an overloaded method is called with the standard progression or its range inheritor's argument
```

Reply:
The argument here **is** a progression or a range, and not **belongs** to them.

Reply:
It was not clear that `argument` is about both progression and range inheritor.